cds_ils/migrator/api.py

```diff
@@ -250,15 +264,232 @@ def import_items_from_json(dump_file, include, rectype="item"):
                     continue
                 try:
                     clean_item_record(record)
-                except ItemMigrationError:
+                except ItemMigrationError as e:
+                    click.secho(str(e), fg="red")
                     continue
-                record = import_record(
-                    record, model, provider, legacy_id_key="barcode"
+                try:
+                    # check if the item already there
+                    item = get_item_by_barcode(record["barcode"])
+                    if item:
+                        click.secho(
+                            "Item {0}) already exists with pid: {1}".format(
+                                record["barcode"], item.pid
+                            ),
+                            fg="blue",
+                        )
+                        continue
+                except ItemMigrationError:
+                    record = import_record(
+                        record, model, provider, legacy_id_key="barcode"
+                    )
+                try:
+                    # without this script is very slow
+                    db.session.commit()
+                except:
+                    db.session.rollback()
+
+
+def import_users_from_json(dump_file):
+    """Imports additional user data from JSON."""
+    dump_file = dump_file[0]
+    with click.progressbar(json.load(dump_file)) as bar:
+        for record in bar:
+            click.echo(
+                'Importing user "{0}({1})"...'.format(
+                    record["id"], record["email"]
                 )
-                if record:
-                    records.append(record)
-    # Index all new item records
-    bulk_index_records(records)
+            )
+            user = get_user_by_person_id(record["ccid"])
+            if not user:
+                click.secho(
+                    "User {0}({1}) not synced via LDAP".format(
+                        record["id"], record["email"]
+                    ),
+                    fg="red",
+                )
+                continue
+                # todo uncomment when more data
+                # raise UserMigrationError
+            else:
+                client_id = current_app.config["CERN_APP_OPENID_CREDENTIALS"][
```

Comment:
does it make sense to refactor the `import_users` function (from LDAP module), and re-use it here? If we fix a bug with the user creation/import from LDAP, we need to remember to fix it here too.

Reply:
actually I'm doing it in this way, because I rely on the users already existing in the db, synced from LDAP and I am adding only the legacy ID. I don't think it should depend on each other, since these data is from 2 "different" sources